File: 1987B/sol.py
def solve():
    _ = int(input())
    a = list(map(int, input().split()))

    curr_max = a[0]
    diff_max = 0
    coins = 0

    for i in a:
        if i < curr_max:
            coins += curr_max - i
            diff_max = max(diff_max, curr_max - i)

        curr_max = max(curr_max, i)

    print(coins + diff_max)


t = int(input())
while t:
    t -= 1
    solve()

File: 1987B/sol.cpp
#include <bits/stdc++.h>
using namespace std;

void solve() {
    int n;
    cin >> n;

    vector<int> a(n);
    for (auto &i : a)
        cin >> i;

    long long coins = 0;
    int max_diff = 0;
    int curr_max = a[0];
    for (int i = 0; i < n; i++) {

        if (a[i] < curr_max) {
            coins += curr_max - a[i];
            max_diff = max(curr_max - a[i], max_diff);
        }
        curr_max = max(curr_max, a[i]);
    }

    cout << coins + max_diff << endl;
}

int main() {
    int t;
    cin >> t;
    while (t--)
        solve();
}
